components/org.wso2.carbon.identity.recovery/src/test/java/org/wso2/carbon/identity/recovery/util/UtilsTest.java

```diff
@@ -0,0 +1,125 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.common.testng.WithCarbonHome;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryClientException;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+import org.wso2.carbon.identity.recovery.internal.IdentityRecoveryServiceDataHolder;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+
+@WithCarbonHome
+public class UtilsTest {
+
+    private static final String TENANT_DOMAIN = "test.com";
+    private static final int TENANT_ID = 123;
+    private static final String USER_NAME = "testUser";
+    private static final String USER_STORE_DOMAIN = "TEST";
+    @Mock
+    private UserStoreManager userStoreManager;
+    @Mock
+    private UserRealm userRealm;
+    @Mock
+    private RealmService realmService;
+    @Mock
+    private IdentityRecoveryServiceDataHolder identityRecoveryServiceDataHolder;
+
+    private MockedStatic<IdentityTenantUtil> mockedStaticIdentityTenantUtil;
+    private MockedStatic<UserStoreManager> mockedStaticUserStoreManager;
+    private MockedStatic<IdentityRecoveryServiceDataHolder> mockedIdentityRecoveryServiceDataHolder;
+
+    @BeforeMethod
+    public void setUp() {
+
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @BeforeClass
+    public void beforeTest() {
+
+        mockedStaticIdentityTenantUtil = mockStatic(IdentityTenantUtil.class);
+        mockedStaticUserStoreManager = mockStatic(UserStoreManager.class);
+        mockedIdentityRecoveryServiceDataHolder = Mockito.mockStatic(IdentityRecoveryServiceDataHolder.class);
+    }
+
+    @AfterClass
+    public void afterTest() {
+
+        mockedStaticIdentityTenantUtil.close();
+        mockedStaticUserStoreManager.close();
+        mockedIdentityRecoveryServiceDataHolder.close();
+    }
+
+    @DataProvider(name = "CheckPasswordPatternViolation")
+    public Object[][] CheckPasswordPatternViolation() {
+
+        return new Object[][]{
+                {new UserStoreException("Invalid Domain Name"), createUser()}
+        };
+    }
+
+    private static User createUser() {
+
+        User user = new User();
+        user.setUserName(USER_NAME);
+        user.setTenantDomain(TENANT_DOMAIN);
+        user.setUserStoreDomain(USER_STORE_DOMAIN);
+        return user;
+    }
+
+    @Test(dataProvider = "CheckPasswordPatternViolation", expectedExceptions = IdentityRecoveryClientException.class)
+    public void testCheckPasswordPatternViolation(UserStoreException exception, User user) throws Exception {
```

Comment:
Since this test is implemented only to test the invalid domain behaviour and as it does not cover the functionality of the `CheckPasswordPatternViolation` method, shall we have a different method name? Ex: `testCheckPasswordPatternViolationForInvalidDomain`